rp2040-hal/src/gpio/mod.rs

```diff
@@ -743,6 +743,17 @@ impl<I: PinId, F: func::Function, P: PullType> Pin<I, F, P> {
             }
         }
     }
+
+    /// Return a wrapper that implements InputPin.
+    ///
+    /// This allows to read from the pin independen from the selected function.
+    /// Depending on the pad configuration, reading from the pin may not return a
+    /// meaninful result. Check the datasheet.
```

Comment:
Nitpick:
```suggestion
    /// This allows to read from the pin independent of the selected function.
    /// Depending on the pad configuration, reading from the pin may not return a
    /// meaningful result.
```
I removed "Check the datasheet" because it's  vague and not actionable - what are they checking for? where should they check?
I couldn't find anything obvious in the datasheet that could tell me when a particular mode would yield a meaningful result, so I would opt to just not say anything about it here.

Reply:
The most important information, IMHO, is this:
"Software control of GPIO, from the single-cycle IO (SIO) block. The SIO function (F5)
must be selected for the processors to drive a GPIO, but the input is always connected,
so software can check the state of GPIOs at any time."
(section 1.4.3, table 3)
So in general, the inputs can be read independent from the configured pin function.

Then there is the input enable flag in the pad control register. You are right, I didn't find any explicit description what happens when that flag isn't set. But from what's written there, I'd assume that the bit must be set to be able to read a meaningful input value.

In any case, I agree with removing "Check the datasheet": We already mention the pad configuration, and these words don't add any useful information.